src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorConfiguration.cs

```diff
@@ -35,8 +39,21 @@ public override int GetHashCode()
         hash.Add(LanguageVersion);
         hash.Add(ConfigurationName);
         hash.Add(Extensions);
+        hash.Add(SuppressAddComponentParameter);
         hash.Add(UseConsolidatedMvcViews);
         hash.Add(LanguageServerFlags);
         return hash;
     }
+
+    public static bool ComputeSuppressAddComponentParameter(Compilation compilation)
+    {
+        var isAddComponentParameterAvailable = compilation
+                                                .GetTypesByMetadataName("Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder")
+                                                .Any(static t =>
+                                                    t.DeclaredAccessibility == Accessibility.Public &&
+                                                    t.GetMembers("AddComponentParameter")
+                                                        .Any(static m => m.DeclaredAccessibility == Accessibility.Public));
```

Comment:
Should we share some of this logic with the other place?

https://github.com/dotnet/razor/blob/bc0b702171653667ce651cdeb3cf844e1b63602f/src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.cs#L238-L249

Or consider adding a comment noting where this logic is duplicated.

Reply:
Moved